tests/phpunit/tests/term.php

```diff
@@ -308,4 +308,52 @@ public function test_orphan_category() {
 		$cat_id2 = self::factory()->category->create( array( 'parent' => $cat_id1 ) );
 		$this->assertWPError( $cat_id2 );
 	}
+
+
+	/**
+	 * @ticket 58329
+	 *
+	 * @dataProvider data_get_term_filter
+	 *
+	 * @param string $filter Term to filter.
+	 */
+	public function test_get_term_filter( $filter ) {
+		$cat_id1 = self::factory()->category->create();
+
+		$term = get_term( $cat_id1, '', OBJECT, $filter );
+
+		$this->assertSame( $filter, $term->filter, 'Sanitize should match' );
+	}
+
+	/**
+	 * @ticket 58329
+	 *
+	 * @dataProvider data_get_term_filter
+	 *
+	 * @param string $filter Term to filter.
+	 */
+	public function test_get_term_filtered( $filter ) {
+		$cat_id1 = self::factory()->category->create();
+		$cat     = self::factory()->category->create_and_get();
+		add_filter(
+			'get_term',
+			static function () use ( $cat ) {
+				return $cat;
+			}
+		);
+
+		$term = get_term( $cat_id1, '', OBJECT, $filter );
+
+		$this->assertSame( $filter, $term->filter, 'Sanitize should match' );
```

Comment:
```suggestion
		$this->assertSame( $filter, $term->filter, "The term's 'filter' property should be set to '$filter'." );
```
Same as [this comment](https://github.com/WordPress/wordpress-develop/pull/5260/files#r1331575257).